mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,20 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.removeOrder = (req, res) => { 
+  Order.findByIdAndDelete(req.params.orderId) 
+    .then(deletedOrder => { 
+      if (!deletedOrder) {
+        return res.status(404).json({
+          error: "Order not found", 
+        });
+      }
+      res.json({ message: "Order deleted successfully" }); 
+    })
```

Comment:
In the `exports.updateOrderStatus` function, the `req.body.orderId` is used to update the order status, but it is not validated or sanitized. This could lead to potential security vulnerabilities. It's recommended to validate and sanitize the input data before using it in the database query.

In the `exports.removeOrder` function, the `req.params.orderId` is used to delete the order, but it is not validated or sanitized. This could lead to potential security vulnerabilities. It's recommended to validate and sanitize the input data before using it in the database query.

In the `exports.create` function, the `req.body.order.user` is assigned to `req.profile`, but it's not clear what `req.profile` is or where it comes from. It's recommended to clarify the origin and purpose of `req.profile`.

In the `exports.listOrders` function, the `Order.find()` method is used to retrieve all orders, but it's not clear what the purpose of this function is or how it's used in the application. It's recommended to add a comment or documentation to explain the purpose of this function.

In the `exports.getStatusValues` function, the `Order.schema.path('status').enumValues` is used to retrieve the status values, but it's not clear what the purpose of this function is or how it's used in the application. It's recommended to add a comment or documentation to explain the purpose of this function.